src/main/kotlin/org/rust/ide/refactoring/extractFunction/RsExtractFunctionConfig.kt

```diff
@@ -222,6 +223,17 @@ class RsExtractFunctionConfig private constructor(
 
     companion object {
         fun create(file: PsiFile, start: Int, end: Int): RsExtractFunctionConfig? {
+            doCreate(file, start, end)?.let { return it }
+
+            file.findElementAtIgnoreWhitespaceAfter(end - 1)?.let { lastElement ->
+                if (lastElement.elementType in listOf(RsElementTypes.COMMA, RsElementTypes.SEMICOLON)) {
+                    return create(file, start, lastElement.startOffset)
```

Comment:
I'd check `lastElement.startOffset < end` before the recursion call just in case (to avoid infinite recursion), either by a condition or an assertion

Reply:
Oh, I think we don't need recursion here (e.g. it is not allowed to have multiple trailing commas)